psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/TransformerData.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.psse.model.pf.io;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.powsybl.psse.model.PsseException;
+import com.powsybl.psse.model.PsseVersion;
+import com.powsybl.psse.model.io.*;
+import com.powsybl.psse.model.pf.PsseTransformer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+import static com.powsybl.psse.model.io.FileFormat.VALID_DELIMITERS;
+import static com.powsybl.psse.model.pf.io.PowerFlowRecordGroup.*;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+class TransformerData extends AbstractRecordGroup<PsseTransformer> {
+
+    TransformerData() {
+        super(TRANSFORMER);
+        withQuotedFields("ckt", "name", "vecgrp");
+        withReaderWriter(FileFormat.LEGACY_TEXT, new TransformersLegacyText(this));
+        withReaderWriter(FileFormat.JSON, new TransformersJson(this));
+        // Fix mappings in nested beans that Univocity did not recognize from annotations
+        // The same mappings are specified in PsseTransformer class
+        // using a headerTransformer processing for @Nested annotation
+        Map<String, String> mapping = new HashMap<>(12 * 3 + 3);
+        for (int k = 1; k <= 12; k++) {
+            mapping.put("winding1Rates.rate" + k, "wdg1rate" + k);
+            mapping.put("winding2Rates.rate" + k, "wdg2rate" + k);
+            mapping.put("winding3Rates.rate" + k, "wdg3rate" + k);
+        }
+        for (char x = 'a'; x <= 'c'; x++) {
+            mapping.put("winding1Rates.rate" + x, "rat" + x + "1");
+            mapping.put("winding2Rates.rate" + x, "rat" + x + "2");
+            mapping.put("winding3Rates.rate" + x, "rat" + x + "3");
+        }
+        //withAttributesToColumnNamesForBeanWriterProcessor(mapping);
```

Comment:
Do you forgot something or is planned later? The mapping is filled but not used.

Reply:
Sorry, I wrote the code to try to fix an issue with Univocity when using `Rates` as a nested object in `TransformerWinding` that is nested in `Transformer`. Finally I decided to move the `Rates` directly as nested objects in `Transformer` to avoid the problem. I have removed the mapping, it is not needed.